src/frontend/parser/expressions.cc

```diff
@@ -6,105 +6,100 @@
-	}
-
-	expression
-	modular_division_expr(expression&& lhs, expression&& rhs)
-	{
-		return expression(expression::type::ModularDivision, { move(lhs), move(rhs) });
-	}
-
-	expression
-	division_expr(expression&& lhs, expression&& rhs)
-	{
-		return expression(expression::type::Division, { move(lhs), move(rhs) });
-	}
-
-	expression
-	equality_expr(expression&& lhs, expression&& rhs)
-	{
-		return expression(expression::type::Equality, { move(lhs), move(rhs) });
-	}
-
-	expression
-	inequality_expr(expression&& lhs, expression&& rhs)
-	{
-		const auto eq = expression(expression::type::Equality, { move(lhs), move(rhs) });
-		return expression(expression::type::Equality, { eq, expression(0) });
-	}
-
-	expression
-	assignment_expr(expression&& lhs, expression&& rhs)
-	{
-		return move(lhs).assign(move(rhs));
-	}
-
-	expression
-	function_call_expr(std::string&& fun_name, expression&& parameter_list /* = {} */)
-	{
-		auto name = expression(identifier(identifier::type::Function, move(fun_name)));
-		parameter_list.front_add(move(name));
-		return expression(
-			expression::type::FunctionCall,
-			move(parameter_list.operands_copy())
-		);
-	}
-
-	expression
-	sequence_expr(expression&& first)
-	{
-		return expression(expression::type::Sequence, { move(first) });
-	}
+expression number_literal(int num)
+{
+	return expression(num);
+}
+
+expression string_literal(std::string &&str)
+{
+	return expression(move(str));
+}
+
+expression identifier_expr(identifier &&ident)
+{
+	return expression(move(ident));
+}
+
+expression nop_expr()
+{
+	return expression(expression::type::Nop, {});
+}
+
+expression negation_expr(expression &&operand)
+{
+	return expression(expression::type::Negation, {move(operand)});
+}
+
+expression addition_expr(expression &&lhs, expression &&rhs)
+{
+	return expression(expression::type::Addition, {move(lhs), move(rhs)});
+}
+
+expression multiplication_expr(expression &&lhs, expression &&rhs)
+{
+	return expression(expression::type::Multiplication, {move(lhs), move(rhs)});
+}
+
+expression modular_division_expr(expression &&lhs, expression &&rhs)
+{
+	return expression(expression::type::ModularDivision, {move(lhs), move(rhs)});
+}
+
+expression division_expr(expression &&lhs, expression &&rhs)
+{
+	return expression(expression::type::Division, {move(lhs), move(rhs)});
+}
 
+expression equality_expr(expression &&lhs, expression &&rhs)
+{
+	return expression(expression::type::Equality, {move(lhs), move(rhs)});
 }
 
-#endif //_KIVI_PARSER_EXPRESSIONS_HH
\ No newline at end of file
+expression inequality_expr(expression &&lhs, expression &&rhs)
+{
+	const auto eq = expression(expression::type::Equality, {move(lhs), move(rhs)});
+	return expression(expression::type::Equality, {eq, expression(0)});
+}
+
+expression assignment_expr(expression &&lhs, expression &&rhs)
+{
+	return move(lhs).assign(move(rhs));
+}
+
+expression function_call_expr(std::string &&fun_name, const std::vector<std::string> &declared_functions,
```

Comment:
It doesn't catch if we're calling a function with incorrect number of parameters or not. This should be taken with much higher priority over the other features of the last milestone.

Reply:
That's pretty interesting since a made a couple of tests on each of the example files. However, I agree with you that this actually is a huge problem and should be positioned as a high priority for the following milestone.

Reply:
The bug has been resolved. Check commit #587be22 for additional info.